core/src/main/java/io/neow3j/crypto/Sign.java

```diff
@@ -181,6 +169,67 @@ public static ECPublicKey recoverFromSignature(int recId, ECDSASignature sig, by
         return new ECPublicKey(q);
     }
 
+    /**
+     * Given an arbitrary piece of text and an NEO message signature encoded in bytes, returns the public key that
+     * was used to sign it. This can then be compared to the expected public key to determine if the signature was
+     * correct.
+     *
+     * @param message       the encoded message.
+     * @param signatureData the message signature components.
+     * @return the public key used to sign the message.
+     * @throws SignatureException if the public key could not be recovered or if there was a signature format error.
```

Comment:
Sorry to be so picky, but the docs are a bit confusing to me. I would simplify to something like "Recovers the public key corresponding to the given signature and message."

The first `@param ` says "encoded message" is it a special encoding? if not then I would simplify to "the message".

Reply:
Agree with @csmuller.